source/crud/tests/README.md

```diff
@@ -737,3 +737,10 @@ that `firstEvent.operationId` is equal to `secondEvent.operationId`. Assert both
 To force completion of the `w:0` writes, execute `coll.countDocuments` and expect the returned count is
 `maxMessageSizeBytes / maxBsonObjectSize + 1`. This is intended to avoid incomplete writes interfering with other tests
 that may use this collection.
+
+### 16. Generated document identifiers are the first field in their document
+
+Construct a `MongoClient` (referred to as `client`) with
+[command monitoring](../../command-logging-and-monitoring/command-logging-and-monitoring.md) enabled to observe
+CommandStartedEvents. Perform an `insert` command using `client` and assert that one CommandStartedEvent (referred to as
+`event`) was observed, and that the first field of `event.command.documents[0]` is `_id`.
```

Comment:
Yea I think we should list out the other APIs to check this for as well.

Should be: insertOne, insertMany, updateOne, updateMany, replaceOne, legacy bulkWrite and client bulkWrite.

Reply:
See my follow up, because some of these APIs need the user to supply the _id the driver does not generate it (update)

Reply:
If an API requires the user to supply `_id`, it should be excluded from this test. Maybe we should have a second test to verify that existing `_id` fields behave as expected, depending on if we decide to re-order or not?

Reply:
@nbbeeken: I don't think any update operations are relevant here, as any generated ID would be client-side for an `upsert` operation. And in that case, we already have several spec tests that specify `_id` in a filter with `upsert: true` and assert that the client-supplied `_id` is relayed properly in the `upsertedIds` response field from the server.

See: "BulkWrite with replaceOne operations" in `bulkWrite.yml` some other tests in `client-bulkWrite-results.yml`.

Reply:
Agreed, learned that after I dove into writing tests for this 😅. I also wrote tests to check that my driver _does not_ reorder `_id`s provided by the user. For that second case update/replace does apply, if we feel we want that coverage as well.